utils/genesis/src/lib.rs

```diff
@@ -89,17 +90,27 @@ where
 
 /// Import a chain from a CAR file. If the snapshot boolean is set, it will not verify the chain
 /// state and instead accept the largest height as genesis.
-pub async fn import_chain<V: ProofVerifier, R: Read, DB>(
+pub async fn import_chain<V: ProofVerifier, DB>(
     sm: &Arc<StateManager<DB>>,
-    reader: R,
+    path: String,
     validate_height: Option<i64>,
 ) -> Result<(), Box<dyn std::error::Error>>
 where
     DB: BlockStore + Send + Sync + 'static,
 {
+    let is_remote_file: bool = path.starts_with("http://") || path.starts_with("https://");
+
     info!("Importing chain from snapshot");
     // start import
-    let cids = load_car(sm.blockstore(), reader)?;
+    let cids = if is_remote_file {
+        let reader = make_reader(path)?;
+        info!("Downloading file...");
+        load_car(sm.blockstore(), reader)?
+    } else {
+        let reader = File::open(&path).expect("Snapshot file path not found!");
+        load_car(sm.blockstore(), reader)?
```

Comment:
Can you wrap the reader in `BufReader`? It's a lot better for reading files

Reply:
updated